### PR TITLE
[2.3] pap,papstatus:  changes to allow Imagewriters to print, and nicer status messages

### PR DESCRIPTION
**Pull request by nutzman**
Based on code from kakwa - https://github.com/kakwa/image-writer/tree/master/bin/pap

### CODE REVIEW

File: bin/pap/pap.c [hunk 1]
```diff
@@ -29,9 +29,11 @@
 
 #define _PATH_PAPRC	".paprc"
 
+#define IMAGEWRITER "ImageWriter"
```
Comment (NJRoadfan) [inline bin/pap/pap.c:32]: This will not work with the ImageWriter LQ as its NBP Type is "LQ".

**Review by NJRoadfan — CHANGES_REQUESTED** [review]
The papstatus program needs to be updated with conditionals checking for NBP type of "ImageWriter" or "LQ" and their special status bytes be processed when one is detected. Normally LaserWriters send back a 255 character Pascal style string with a status message, while the ImageWriters send back a raw 2-byte value.

File: bin/pap/pap.c [hunk 2]
```diff
@@ -163,6 +166,8 @@ int main( int ac, char	**av)
     extern char		*optarg;
     extern int		optind;
 
+	int		is_imagewriter;
```
Comment (rdmark) [inline bin/pap/pap.c:169]: Nitpick, but indentation should be four spaces and not a tab here.

In fairness, the indentation can be a bit all over the place in this codebase. But let's use the same indentation as neighboring code to maintain readability.

File: bin/pap/pap.c [hunk 3]
```diff
@@ -255,6 +260,9 @@ int main( int ac, char	**av)
 	exit( 2 );
     }
 
+	is_imagewriter = (0 == strcmp(IMAGEWRITER, type) ||
+		0 == strcmp(IMAGEWRITER_LQ, type)  );
```
Comment (rdmark) [inline bin/pap/pap.c:264]: indentation

File: bin/pap/pap.c [hunk 4]
```diff
@@ -804,14 +812,16 @@ static int send_file( int fd, ATP atp, int lastfile)
 	    /*
 	     * The stinking LaserWriter IINTX puts crap in this
 	     * field.
+		 * 
+		 * The Imagewriter II doesn't handle this well, either
```
Comment (rdmark) [inline bin/pap/pap.c:816]: indentation

File: bin/pap/papstatus.c [hunk 5]
```diff
@@ -146,23 +168,26 @@ int main( int ac, char **av)
 	exit( 1 );
     }
 
+	is_imagewriter = (0 == strcmp(IMAGEWRITER, type) ||
+		0 == strcmp(IMAGEWRITER_LQ, type)  );
```
Comment (rdmark) [inline bin/pap/papstatus.c:172]: indentation

**Comment by nutzman** [comment]
I updated my changes to use spaces. I could format the entire files if you like.

**Review by rdmark — APPROVED** [review]

**Comment by rdmark** [comment]
Thanks again for your contribution! My HP DeskWriter 550C with LocalTalk / PAP support was delivered yesterday, but I didn't have a chance to unpack and set it up yet. I'm looking forward to playing around with `pap` and `papstatus` soon!

**Pull request merged**